src/mscorlib/src/System/String.cs

```diff
@@ -957,9 +975,31 @@ public String[] Split(char[] separator, int count, StringSplitOptions options)
             return SplitInternal(separator, count, options);
         }
 
+        [System.Security.SecuritySafeCritical]
+        private unsafe String[] SplitInternal(char separator, int count, StringSplitOptions options)
+        {
+            char* pSeparators = stackalloc char[1];
+            pSeparators[0] = separator;
+            return SplitInternal(pSeparators, /*separatorsLength*/ 1, count, options);
+        }
+
         [ComVisible(false)]
+        [System.Security.SecuritySafeCritical]
         internal String[] SplitInternal(char[] separator, int count, StringSplitOptions options)
         {
+            unsafe
```

Comment:
The flip side is [this](https://twitter.com/jaredpar/status/468434410568577025):

> If unsafe is an implementation detail then why mark the methods unsafe?

A moot point, but I digress.